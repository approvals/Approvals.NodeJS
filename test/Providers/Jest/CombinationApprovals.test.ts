import {describe, test} from "@jest/globals";
import {
    verifyAllCombinations1,
    verifyAllCombinations2,
    verifyAllCombinations3,
    verifyAllCombinations4,
    verifyAllCombinations5,
    verifyAllCombinations6,
    verifyAllCombinations7, verifyAllCombinations8, verifyAllCombinations9
} from "../../../lib/Providers/Jest/CombinationApprovals";


describe("Combinations", () => {
    test("verify1", () => {
        function binarySquare(i: number) {
            if (i === 2) {
                throw new Error("There is no 2");
            }
            return i;
        }

        const numbers = [0, 1, 2];
        verifyAllCombinations1(binarySquare, numbers);
    });
    test("verify2", () => {
        const numbers = [0, 1, 2];
        verifyAllCombinations2((a, b) => a + b, numbers, numbers);
    });
    test("verify3", () => {
        const numbers = [3, 4];
        verifyAllCombinations3((a, b, c) => a * b * c, numbers, numbers, numbers);
    });
    test("verify4", () => {
        const numbers = [3, 4];
        verifyAllCombinations4((a, b, c, d) => a + "" + b + c + d, numbers, numbers, numbers, numbers);
    });
    test("verify5", () => {
        verifyAllCombinations5((a, b, c, d, e) => e + d + c + b + a, ["a", "b"], ["a", "b"], ["c", "d"], ["e", "f"], ["g", "h", "i"]);
    });
    test("verify6", () => {
        verifyAllCombinations6((a, b, c, d, e, f) => "#" + a + " " + f + e + d + c + b, [1, 2], ["a", "b"], ["c", "d"], ["e", "f"], ["i", "j"], ["g", "h"]);
    });
    test("verify7", () => {
        verifyAllCombinations7((a, b, c, d, e, f, g) => a + b + c + d + e + f + g, [1, 2], [2, 3], [2, 3], [2, 3], [2, 3], [2, 3], [2, 3]);
    });
    test("verify8", () => {
        verifyAllCombinations8((a, b, c, d, e, f, g, h) => a + b + c + d + e + f + g + h, [1, 2], [2, 3], [2, 3], [2, 3], [2, 3], [2, 3], [2, 3], [8, 7]);
    });
    test("verify9", () => {
        verifyAllCombinations9((a, b, c, d, e, f, g, h, i) => "" + a + b + c + d + e + f + g + h + i, [1, 2], [2, 3], [2, 3], [2, 3], [2, 3], [2, 3], [2, 3], [8, 7], [9, 8]);
    });
});

describe('documentation', () => {
    // begin-snippet: CombinationsStartingPoint
    test('CombinationsStartingPoint', () => {
        const inputs1 = ["input1.value1", "input1.value2"];
        const inputs2 = ["input2.value1", "input2.value2", "input2.value3"];
        verifyAllCombinations2((a, b) => `placeholder for ${a}, ${b}`, inputs1, inputs2);
    });
    // end-snippet
});
