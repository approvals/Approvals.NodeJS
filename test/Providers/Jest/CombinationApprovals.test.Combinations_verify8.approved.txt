[1,2,2,2,2,2,2,8] => 21
[1,2,2,2,2,2,2,7] => 20
[1,2,2,2,2,2,3,8] => 22
[1,2,2,2,2,2,3,7] => 21
[1,2,2,2,2,3,2,8] => 22
[1,2,2,2,2,3,2,7] => 21
[1,2,2,2,2,3,3,8] => 23
[1,2,2,2,2,3,3,7] => 22
[1,2,2,2,3,2,2,8] => 22
[1,2,2,2,3,2,2,7] => 21
[1,2,2,2,3,2,3,8] => 23
[1,2,2,2,3,2,3,7] => 22
[1,2,2,2,3,3,2,8] => 23
[1,2,2,2,3,3,2,7] => 22
[1,2,2,2,3,3,3,8] => 24
[1,2,2,2,3,3,3,7] => 23
[1,2,2,3,2,2,2,8] => 22
[1,2,2,3,2,2,2,7] => 21
[1,2,2,3,2,2,3,8] => 23
[1,2,2,3,2,2,3,7] => 22
[1,2,2,3,2,3,2,8] => 23
[1,2,2,3,2,3,2,7] => 22
[1,2,2,3,2,3,3,8] => 24
[1,2,2,3,2,3,3,7] => 23
[1,2,2,3,3,2,2,8] => 23
[1,2,2,3,3,2,2,7] => 22
[1,2,2,3,3,2,3,8] => 24
[1,2,2,3,3,2,3,7] => 23
[1,2,2,3,3,3,2,8] => 24
[1,2,2,3,3,3,2,7] => 23
[1,2,2,3,3,3,3,8] => 25
[1,2,2,3,3,3,3,7] => 24
[1,2,3,2,2,2,2,8] => 22
[1,2,3,2,2,2,2,7] => 21
[1,2,3,2,2,2,3,8] => 23
[1,2,3,2,2,2,3,7] => 22
[1,2,3,2,2,3,2,8] => 23
[1,2,3,2,2,3,2,7] => 22
[1,2,3,2,2,3,3,8] => 24
[1,2,3,2,2,3,3,7] => 23
[1,2,3,2,3,2,2,8] => 23
[1,2,3,2,3,2,2,7] => 22
[1,2,3,2,3,2,3,8] => 24
[1,2,3,2,3,2,3,7] => 23
[1,2,3,2,3,3,2,8] => 24
[1,2,3,2,3,3,2,7] => 23
[1,2,3,2,3,3,3,8] => 25
[1,2,3,2,3,3,3,7] => 24
[1,2,3,3,2,2,2,8] => 23
[1,2,3,3,2,2,2,7] => 22
[1,2,3,3,2,2,3,8] => 24
[1,2,3,3,2,2,3,7] => 23
[1,2,3,3,2,3,2,8] => 24
[1,2,3,3,2,3,2,7] => 23
[1,2,3,3,2,3,3,8] => 25
[1,2,3,3,2,3,3,7] => 24
[1,2,3,3,3,2,2,8] => 24
[1,2,3,3,3,2,2,7] => 23
[1,2,3,3,3,2,3,8] => 25
[1,2,3,3,3,2,3,7] => 24
[1,2,3,3,3,3,2,8] => 25
[1,2,3,3,3,3,2,7] => 24
[1,2,3,3,3,3,3,8] => 26
[1,2,3,3,3,3,3,7] => 25
[1,3,2,2,2,2,2,8] => 22
[1,3,2,2,2,2,2,7] => 21
[1,3,2,2,2,2,3,8] => 23
[1,3,2,2,2,2,3,7] => 22
[1,3,2,2,2,3,2,8] => 23
[1,3,2,2,2,3,2,7] => 22
[1,3,2,2,2,3,3,8] => 24
[1,3,2,2,2,3,3,7] => 23
[1,3,2,2,3,2,2,8] => 23
[1,3,2,2,3,2,2,7] => 22
[1,3,2,2,3,2,3,8] => 24
[1,3,2,2,3,2,3,7] => 23
[1,3,2,2,3,3,2,8] => 24
[1,3,2,2,3,3,2,7] => 23
[1,3,2,2,3,3,3,8] => 25
[1,3,2,2,3,3,3,7] => 24
[1,3,2,3,2,2,2,8] => 23
[1,3,2,3,2,2,2,7] => 22
[1,3,2,3,2,2,3,8] => 24
[1,3,2,3,2,2,3,7] => 23
[1,3,2,3,2,3,2,8] => 24
[1,3,2,3,2,3,2,7] => 23
[1,3,2,3,2,3,3,8] => 25
[1,3,2,3,2,3,3,7] => 24
[1,3,2,3,3,2,2,8] => 24
[1,3,2,3,3,2,2,7] => 23
[1,3,2,3,3,2,3,8] => 25
[1,3,2,3,3,2,3,7] => 24
[1,3,2,3,3,3,2,8] => 25
[1,3,2,3,3,3,2,7] => 24
[1,3,2,3,3,3,3,8] => 26
[1,3,2,3,3,3,3,7] => 25
[1,3,3,2,2,2,2,8] => 23
[1,3,3,2,2,2,2,7] => 22
[1,3,3,2,2,2,3,8] => 24
[1,3,3,2,2,2,3,7] => 23
[1,3,3,2,2,3,2,8] => 24
[1,3,3,2,2,3,2,7] => 23
[1,3,3,2,2,3,3,8] => 25
[1,3,3,2,2,3,3,7] => 24
[1,3,3,2,3,2,2,8] => 24
[1,3,3,2,3,2,2,7] => 23
[1,3,3,2,3,2,3,8] => 25
[1,3,3,2,3,2,3,7] => 24
[1,3,3,2,3,3,2,8] => 25
[1,3,3,2,3,3,2,7] => 24
[1,3,3,2,3,3,3,8] => 26
[1,3,3,2,3,3,3,7] => 25
[1,3,3,3,2,2,2,8] => 24
[1,3,3,3,2,2,2,7] => 23
[1,3,3,3,2,2,3,8] => 25
[1,3,3,3,2,2,3,7] => 24
[1,3,3,3,2,3,2,8] => 25
[1,3,3,3,2,3,2,7] => 24
[1,3,3,3,2,3,3,8] => 26
[1,3,3,3,2,3,3,7] => 25
[1,3,3,3,3,2,2,8] => 25
[1,3,3,3,3,2,2,7] => 24
[1,3,3,3,3,2,3,8] => 26
[1,3,3,3,3,2,3,7] => 25
[1,3,3,3,3,3,2,8] => 26
[1,3,3,3,3,3,2,7] => 25
[1,3,3,3,3,3,3,8] => 27
[1,3,3,3,3,3,3,7] => 26
[2,2,2,2,2,2,2,8] => 22
[2,2,2,2,2,2,2,7] => 21
[2,2,2,2,2,2,3,8] => 23
[2,2,2,2,2,2,3,7] => 22
[2,2,2,2,2,3,2,8] => 23
[2,2,2,2,2,3,2,7] => 22
[2,2,2,2,2,3,3,8] => 24
[2,2,2,2,2,3,3,7] => 23
[2,2,2,2,3,2,2,8] => 23
[2,2,2,2,3,2,2,7] => 22
[2,2,2,2,3,2,3,8] => 24
[2,2,2,2,3,2,3,7] => 23
[2,2,2,2,3,3,2,8] => 24
[2,2,2,2,3,3,2,7] => 23
[2,2,2,2,3,3,3,8] => 25
[2,2,2,2,3,3,3,7] => 24
[2,2,2,3,2,2,2,8] => 23
[2,2,2,3,2,2,2,7] => 22
[2,2,2,3,2,2,3,8] => 24
[2,2,2,3,2,2,3,7] => 23
[2,2,2,3,2,3,2,8] => 24
[2,2,2,3,2,3,2,7] => 23
[2,2,2,3,2,3,3,8] => 25
[2,2,2,3,2,3,3,7] => 24
[2,2,2,3,3,2,2,8] => 24
[2,2,2,3,3,2,2,7] => 23
[2,2,2,3,3,2,3,8] => 25
[2,2,2,3,3,2,3,7] => 24
[2,2,2,3,3,3,2,8] => 25
[2,2,2,3,3,3,2,7] => 24
[2,2,2,3,3,3,3,8] => 26
[2,2,2,3,3,3,3,7] => 25
[2,2,3,2,2,2,2,8] => 23
[2,2,3,2,2,2,2,7] => 22
[2,2,3,2,2,2,3,8] => 24
[2,2,3,2,2,2,3,7] => 23
[2,2,3,2,2,3,2,8] => 24
[2,2,3,2,2,3,2,7] => 23
[2,2,3,2,2,3,3,8] => 25
[2,2,3,2,2,3,3,7] => 24
[2,2,3,2,3,2,2,8] => 24
[2,2,3,2,3,2,2,7] => 23
[2,2,3,2,3,2,3,8] => 25
[2,2,3,2,3,2,3,7] => 24
[2,2,3,2,3,3,2,8] => 25
[2,2,3,2,3,3,2,7] => 24
[2,2,3,2,3,3,3,8] => 26
[2,2,3,2,3,3,3,7] => 25
[2,2,3,3,2,2,2,8] => 24
[2,2,3,3,2,2,2,7] => 23
[2,2,3,3,2,2,3,8] => 25
[2,2,3,3,2,2,3,7] => 24
[2,2,3,3,2,3,2,8] => 25
[2,2,3,3,2,3,2,7] => 24
[2,2,3,3,2,3,3,8] => 26
[2,2,3,3,2,3,3,7] => 25
[2,2,3,3,3,2,2,8] => 25
[2,2,3,3,3,2,2,7] => 24
[2,2,3,3,3,2,3,8] => 26
[2,2,3,3,3,2,3,7] => 25
[2,2,3,3,3,3,2,8] => 26
[2,2,3,3,3,3,2,7] => 25
[2,2,3,3,3,3,3,8] => 27
[2,2,3,3,3,3,3,7] => 26
[2,3,2,2,2,2,2,8] => 23
[2,3,2,2,2,2,2,7] => 22
[2,3,2,2,2,2,3,8] => 24
[2,3,2,2,2,2,3,7] => 23
[2,3,2,2,2,3,2,8] => 24
[2,3,2,2,2,3,2,7] => 23
[2,3,2,2,2,3,3,8] => 25
[2,3,2,2,2,3,3,7] => 24
[2,3,2,2,3,2,2,8] => 24
[2,3,2,2,3,2,2,7] => 23
[2,3,2,2,3,2,3,8] => 25
[2,3,2,2,3,2,3,7] => 24
[2,3,2,2,3,3,2,8] => 25
[2,3,2,2,3,3,2,7] => 24
[2,3,2,2,3,3,3,8] => 26
[2,3,2,2,3,3,3,7] => 25
[2,3,2,3,2,2,2,8] => 24
[2,3,2,3,2,2,2,7] => 23
[2,3,2,3,2,2,3,8] => 25
[2,3,2,3,2,2,3,7] => 24
[2,3,2,3,2,3,2,8] => 25
[2,3,2,3,2,3,2,7] => 24
[2,3,2,3,2,3,3,8] => 26
[2,3,2,3,2,3,3,7] => 25
[2,3,2,3,3,2,2,8] => 25
[2,3,2,3,3,2,2,7] => 24
[2,3,2,3,3,2,3,8] => 26
[2,3,2,3,3,2,3,7] => 25
[2,3,2,3,3,3,2,8] => 26
[2,3,2,3,3,3,2,7] => 25
[2,3,2,3,3,3,3,8] => 27
[2,3,2,3,3,3,3,7] => 26
[2,3,3,2,2,2,2,8] => 24
[2,3,3,2,2,2,2,7] => 23
[2,3,3,2,2,2,3,8] => 25
[2,3,3,2,2,2,3,7] => 24
[2,3,3,2,2,3,2,8] => 25
[2,3,3,2,2,3,2,7] => 24
[2,3,3,2,2,3,3,8] => 26
[2,3,3,2,2,3,3,7] => 25
[2,3,3,2,3,2,2,8] => 25
[2,3,3,2,3,2,2,7] => 24
[2,3,3,2,3,2,3,8] => 26
[2,3,3,2,3,2,3,7] => 25
[2,3,3,2,3,3,2,8] => 26
[2,3,3,2,3,3,2,7] => 25
[2,3,3,2,3,3,3,8] => 27
[2,3,3,2,3,3,3,7] => 26
[2,3,3,3,2,2,2,8] => 25
[2,3,3,3,2,2,2,7] => 24
[2,3,3,3,2,2,3,8] => 26
[2,3,3,3,2,2,3,7] => 25
[2,3,3,3,2,3,2,8] => 26
[2,3,3,3,2,3,2,7] => 25
[2,3,3,3,2,3,3,8] => 27
[2,3,3,3,2,3,3,7] => 26
[2,3,3,3,3,2,2,8] => 26
[2,3,3,3,3,2,2,7] => 25
[2,3,3,3,3,2,3,8] => 27
[2,3,3,3,3,2,3,7] => 26
[2,3,3,3,3,3,2,8] => 27
[2,3,3,3,3,3,2,7] => 26
[2,3,3,3,3,3,3,8] => 28
[2,3,3,3,3,3,3,7] => 27
